project-2-e-commerce/src/component/cart/CartItem.js

```diff
@@ -0,0 +1,83 @@
+import { useDispatch, useSelector } from "react-redux";
+import { formatPrice } from "../../helpers";
+import { updateUserCart, updateLocalCarts, setDelCart } from "../../store/slices/UserSlice"
+
+function CartItem({ item }) {
+    const { name, img, price, quantity, id } = item
+    const priceFormat = formatPrice(price)
+    const totalFormat = formatPrice(price*quantity)
+    const { isLogged, userData, localCarts } = useSelector(({user}) => user)
+    const dispatch = useDispatch()
+
+    const handleChangeQnt = (e) => {
+        let value
+        switch (e) {
+            case 'increment':
+                value = quantity + 1
+                break;
+            case 'decrement':
+                value = quantity > 1 ? quantity - 1 : quantity
+                break;
+            default:
+                value = isNaN(e.target.value) ? 1 : e.target.value
+                break;
+        }
+        if (isLogged) {
+            const currentCarts = userData.cart
+            const newCarts = updateCartList(id, value, currentCarts)
+            dispatch(updateUserCart(newCarts))
+        } else {
+            const newLocalCarts = updateCartList(id, value, localCarts)
+            dispatch(updateLocalCarts(newLocalCarts))
+        }
+    }
+
+    const updateCartList = (id, value, arr) => {
+        return arr.map(cart => {
+            if (cart.id === id) {
+                return {...cart, quantity: value}
+            } else return cart 
+        })
```

Comment:
```suggestion
        return arr.map(cart => {
            if (cart.id === id) {
                return {...cart, quantity: value}
            }
            return cart 
        })
```
Hoặc
`return arr.map(cart => (cart.id === id ? {...cart, quantity: value} : cart))`